vmm/src/acpi.rs

```diff
@@ -626,81 +628,84 @@ fn create_viot_table(iommu_bdf: &PciBdf, devices_bdf: &[PciBdf]) -> Sdt {
     viot
 }
 
-pub fn create_acpi_tables(
-    guest_mem: &GuestMemoryMmap,
+fn create_acpi_tables_internal(
+    dsdt_offset: GuestAddress,
     device_manager: &Arc<Mutex<DeviceManager>>,
     cpu_manager: &Arc<Mutex<CpuManager>>,
     memory_manager: &Arc<Mutex<MemoryManager>>,
     numa_nodes: &NumaNodes,
     tpm_enabled: bool,
-) -> GuestAddress {
+    #[cfg(feature = "fw_cfg")] add_to_fw_cfg: bool,
+) -> (Rsdp, Vec<u8>) {
     trace_scoped!("create_acpi_tables");
-
-    let start_time = Instant::now();
-    let rsdp_offset = arch::layout::RSDP_POINTER;
     let mut tables: Vec<u64> = Vec::new();
 
+    // To create AcpiTable
+    let mut table_bytes: Vec<u8> = vec![];
+    #[cfg(feature = "fw_cfg")]
+    let mut pointers: Vec<usize> = vec![];
+    #[cfg(feature = "fw_cfg")]
+    let mut checksums: Vec<(usize, usize)> = vec![];
```

Comment:
Maybe the naming is confusing, can you elaborate what exactly that `fw_cfg` needs for creating ACPI tables? I kind of see you need a pair of "offset/address" and "size" for some of the tables.